src/main/java/com/bandwidth/sqs/queue/buffer/task_buffer/KeyedTaskBuffer.java

```diff
@@ -11,11 +11,20 @@
 import java.util.Timer;
 import java.util.TimerTask;
 
+import io.reactivex.subjects.SingleSubject;
+
+/**
+ * A buffer that buffers individual data, collecting it in bucket by key, then running a task to batch
+ * process the data
+ *
+ * @param <K> Key
+ * @param <D> Data
+ */
 public class KeyedTaskBuffer<K, D> {
     private Timer timer = new Timer();
     private final int maxBufferSize;
     private final Duration maxWait;
-    private final Map<K, Buffer<D>> buffers = new HashMap<>();
+    private final Map<K, List<D>> buffers = new HashMap<>();
```

Comment:
this looks conspicuously like a `Multimap`, would that make working with this thing easier?
https://github.com/google/guava/wiki/NewCollectionTypesExplained#multimap